src/device_builder.rs

```diff
@@ -106,27 +107,60 @@ impl<'a, B: UsbBus> UsbDeviceBuilder<'a, B> {
         self
     }
 
+    /// Sets **extra** Language ID for device.
+    ///
+    /// Since "en_US"(0x0409) is implicitly embedded, you just need to fill other LangIDs
+    ///
+    /// Default: (none)
+    pub fn extra_lang_ids(mut self, extra_lang_ids: &'a [u16]) -> Self {
+        assert!(
+            extra_lang_ids.len() < 16,
+            "Not support more than 15 extra LangIDs"
+        );
+
+        if extra_lang_ids.len() > 0 {
+            self.config.extra_lang_ids = Some(extra_lang_ids);
+        }
+
+        self
```

Comment:
After this update, check `manufacturer`, `product`, and `serial_number` to ensure they have the correct number of slice entries. The number of extra-lang-ids and the number of elements in these slices must always be equal.

Obviously only do this if those fields are Some (can't check before they're set)

Reply:
If we check lang id count in this function, we assume user set `extra_lang_id` before `manufacturer`, `product`, and `serial_number`. I'm not sure whether we should force/assume user will do this or not.

Reply:
No, we just don't check if `extra_lang_id` is None. We'll check later when they actually specify the extra lang IDs

Reply:
done